### PR TITLE
perf(UI): redesign settings advanced tab

### PR DESCRIPTION
**Pull request by Jerry23011**
### Preview
截屏2024-05-07 02 23 03
截屏2024-05-07 02 24 22

### CODE REVIEW

**Review by AkaShark — DISMISSED** [review]

**Review by tisfeng — DISMISSED** [review]
Looks good.

File: Easydict/Swift/View/AdvancedTabStyle.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+//  Easydict
+//
+//  Created by Jerry on 2024-05-06.
+//  Copyright © 2024 izual. All rights reserved.
+//
+
+import SwiftUI
+
+/// Takes in a Color, a systemImage, a text label, and an optional subtitle to quickly create a toggle or picker style for Advanced Tab in Settings.
+struct AdvancedTabStyle: View {
```
Comment (phlpsong) [inline Easydict/Swift/View/AdvancedTabStyle.swift:12]: The name is a bit confusing, maybe something like `AdvancedTabItemView`.

File: Easydict/Swift/View/AdvancedTabStyle.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                .clipShape(RoundedRectangle(cornerRadius: 4))
+                .overlay(
+                    Image(systemName: systemImage)
+                        .font(.system(size: 12))
+                        .foregroundColor(.white)
+                )
+            VStack(alignment: .leading) {
+                Text(labelText)
+                if let subtitle = subtitleText {
+                    Text(subtitle)
```
Comment (phlpsong) [inline Easydict/Swift/View/AdvancedTabStyle.swift:32]:
```suggestion
                if let subtitleText {
                    Text(subtitleText)
```

File: Easydict/Swift/View/SettingView/Tabs/TabView/AdvancedTab.swift [hunk 3]
```diff
@@ -16,26 +16,45 @@ struct AdvancedTab: View {
...
                         Text(option.localizedStringResource)
                             .tag(option)
                     }
                 }
-                Toggle("setting.general.advance.enable_beta_feature", isOn: $enableBetaFeature)
+            }
+            Section {
+                Toggle(isOn: $enableBetaFeature) {
+                    AdvancedTabStyle(
+                        color: Color.blue,
```
Comment (phlpsong) [inline Easydict/Swift/View/SettingView/Tabs/TabView/AdvancedTab.swift:36]:
```suggestion
                        color: .blue,
```

File: Easydict/Swift/View/SettingView/Tabs/TabView/AdvancedTab.swift [hunk 4]
```diff
@@ -16,26 +16,45 @@ struct AdvancedTab: View {
     var body: some View {
         Form {
             Section {
-                Picker("setting.general.advance.default_tts_service", selection: $defaultTTSServiceType) {
+                Picker(
+                    selection: $defaultTTSServiceType,
+                    label: AdvancedTabStyle(
+                        color: Color.orange,
```
Comment (phlpsong) [inline Easydict/Swift/View/SettingView/Tabs/TabView/AdvancedTab.swift:22]:
```suggestion
                        color: .orange,
```

**Review by phlpsong — CHANGES_REQUESTED** [review]

File: Easydict/Swift/View/AdvancedTabStyle.swift (hunk 1 above)
Comment (Jerry23011) [reply Easydict/Swift/View/AdvancedTabStyle.swift]: Thank you so much for pointing these out

**Review by phlpsong — APPROVED** [review]
LGTM!

**Review by AkaShark — APPROVED** [review]

**Pull request merged**